doc/sonic-application-extension/tpcm_app_ext.md

```diff
@@ -0,0 +1,467 @@
+
+
+<!-- omit in toc -->
+###### Examples
+
+Update the container section 'memory' limit and service section 'after' attribute
+```
+admin@sonic:~$ sudo sonic-package-manager manifests update my_node_exporter --tpcm --container memory 1000M --service after swss
+admin@sonic:~$ sudo sonic-package-manager update my_node_exporter --use-local-manifest 
+```
+
+
+#### TPC Upgrade CLI
+
+No change in the sonic-package-manager CLI for upgrade which is 'sonic-package-manager install'
+
+
+#### TPC Manifests CLI
+
+##### Manifest Create
+
+sonic-package-manager manifests create <manifest-name> --tpcm --container <attribute> <value> --service <attribute> <value>
+
+<!-- omit in toc -->
+###### Examples
+Create a default local tpcm manifest file
+```
+admin@sonic:~$ sudo sonic-package-manager manifests create my_httpd --tpcm 
+```
+
+Create a local tpcm manifest file with container memory 1024M and privileged and flag set to true
+```
+admin@sonic:~$ sudo sonic-package-manager manifests create my_httpd --tpcm --container memory 1024M --container privileged true
+```
+
+
+##### Manifest Update
+
+sonic-package-manager manifests update <manifest-name> --container <attribute> <value> --service <attribute> <value>
+
+<!-- omit in toc -->
+###### Examples
+Update a tpcm manifest file with cpu-quota 30000
+```
+admin@sonic:~$ sudo sonic-package-manager manifests update my_httpd --container cpu-quota 30000
+```
+
+
+##### Manifest show
+
+sonic-package-manager manifests show  <manifest-name>
+
+<!-- omit in toc -->
+###### Examples
+Display the contents of a local tpcm manifest file
+```
+admin@sonic:~$ sudo sonic-package-manager manifests show my_httpd
+```
+
+##### Manifest delete
+
+sonic-package-manager manifests delete  <manifest-name>
+
+<!-- omit in toc -->
+###### Examples
+Delete the local tpcm manifest file
+```
+admin@sonic:~$ sudo sonic-package-manager manifests delete my_httpd
+```
+
+##### Manifest list
+
+sonic-package-manager manifests list
+
+<!-- omit in toc -->
+###### Examples
+Display all the local tpcm manifest files from TPC manifest folder
+```
+admin@sonic:~$ sudo sonic-package-manager manifests list
+```
+
+
+### Future Enhancements Proposal
+
+- Local custom Manfiest file could be used(created/updated) for SONiC packages as well.
+- TPC could be configured to start right after system becomes ready so that SONiC packages bootup wont be delayed.
+- Data preservation in TPCM migration
+- All manifest file properties could be updated. 
+
+
+### Testing Requirements/Design
+
+#### Unit Test cases
+
+Installation test case: 
+- Verify that TPCM packages can be installed successfully using the sonic package manager with a custom local manifest file.
+- Verify with SCP
+- Verify with SFTP
+- Verify with url
+
+Uninstallation test case: 
+- Verify that TPCM packages can be uninstalled using the sonic package manager without any issues.
+
+Upgrade test case: 
```

Comment:
Is SONiC-2-SONiC upgrade considered? What will happen with TPC extension if we upgrade base SONiC image to new version?

Reply:
Added the TC